src/libraries/System.Linq/src/System/Linq/First.cs

```diff
@@ -114,6 +114,19 @@ public static TSource FirstOrDefault<TSource>(this IEnumerable<TSource> source,
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.predicate);
             }
 
+            if (source.TryGetSpan(out ReadOnlySpan<TSource> span))
+            {
+                for (int i = 0; i < span.Length; i++)
+                {
+                    TSource element = span[i];
```

Comment:
```suggestion
                foreach (TSource element in span)
                {
```
AFAIR Roslyn transforms such `foreach`es on `Span`s into `for` loops anyway.